src/torrent.cpp

```diff
@@ -6102,6 +6501,37 @@ bool is_downloading_state(int const st)
 					ret.piece_priorities.push_back(m_picker->piece_priority(i));
 			}
 		}
+
+		if (m_torrent_file->info_hash().has_v2())
```

Comment:
I think there should be a test for the this. It could probably be as simple as loading a merkle tree and verified_leaf_hashes from resume data, then save resume data again and make sure we get back what we put in